src/java_binding/src/lib.rs

```diff
@@ -552,6 +593,125 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetDecimalVal
+        let (class_ref, constructor) =
+            pointer.as_ref().class_cache.date_ctor.get_or_try_init(|| {
+                let cls = env.find_class("java/sql/Date")?;
+                let init_method = env.get_static_method_id(
+                    cls,
+                    "valueOf",
+                    "(Ljava/lang/String;)Ljava/sql/Date;",
+                )?;
+                Ok::<_, jni::errors::Error>((env.new_global_ref(cls)?, init_method))
+            })?;
+        let class = JClass::from(class_ref.as_obj());
+        let JValue::Object(date_obj) = env.call_static_method_unchecked(
+            class,
+            *constructor,
+            ReturnType::Object,
+            &[jvalue::from(JValue::from(string_value))],
+        )? else {
+            return Err(BindingError::from(jni::errors::Error::MethodNotFound {
+                name: "valueOf".to_string(),
+                sig: "(Ljava/lang/String;)Ljava/sql/Date;".into(),
+            }));
+        };
+        Ok(date_obj)
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetTimeValue<'a>(
+    env: EnvParam<'a>,
+    pointer: Pointer<'a, JavaBindingRow>,
+    idx: jint,
+) -> JObject<'a> {
+    execute_and_catch(env, move || {
+        let value = pointer
+            .as_ref()
+            .datum_at(idx as usize)
+            .unwrap()
+            .into_time()
+            .0
+            .to_string();
+
+        let string_value = env.new_string(value)?;
+        let (class_ref, constructor) =
+            pointer.as_ref().class_cache.time_ctor.get_or_try_init(|| {
+                let cls = env.find_class("java/sql/Time")?;
+                let init_method = env.get_static_method_id(
+                    cls,
+                    "valueOf",
+                    "(Ljava/lang/String;)Ljava/sql/Time;",
+                )?;
+                Ok::<_, jni::errors::Error>((env.new_global_ref(cls)?, init_method))
+            })?;
+        let class = JClass::from(class_ref.as_obj());
+        let JValue::Object(obj) = env.call_static_method_unchecked(
+            class,
+            *constructor,
+            ReturnType::Object,
+            &[jvalue::from(JValue::from(string_value))],
+        )? else {
+            return Err(BindingError::from(jni::errors::Error::MethodNotFound {
+                name: "valueOf".to_string(),
+                sig: "(Ljava/lang/String;)Ljava/sql/Time;".into(),
+            }));
+        };
+        Ok(obj)
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetByteaValue<'a>(
+    env: EnvParam<'a>,
+    pointer: Pointer<'a, JavaBindingRow>,
+    idx: jint,
+) -> JObject<'a> {
+    execute_and_catch(env, move || {
+        let bytes = pointer
+            .as_ref()
+            .datum_at(idx as usize)
+            .unwrap()
+            .into_bytea();
+        let bytes_value = env.byte_array_from_slice(bytes)?;
```

Comment:
Any reason to return the `ByteArrayInputStream` rather than a Byte array itself?

Reply:
because we can then use `stmt.setBinaryStream` to initialize the`Bytea` or `Blob` field in the `PreparedStatement`